### PR TITLE
add support for mavlink buttons in the manual control setpoint topic

### PR DESCRIPTION
**Pull request by tstastny**
### Solved Problem
I wanted to trigger some things inside of flight modes with a button (remote controller sending commands via mavlink) but realized we currently have no way of doing this.

### Solution
Mavlink spec already supports 16 buttons (32 with the extension fields, not included in this PR currently). I added the bitfield to the manual control setpoint msg def. 

https://mavlink.io/en/messages/common.html#MANUAL_CONTROL

### Changelog Entry
For release notes:
```
Feature: added button support for mavlink-based manual control
```

### Alternatives
- could also include the extension (buttons2)
- could also rename the field, or translate to separate button fields instead of just copying over the bitmask
- could also keep this entirely separate from mavlink spec, and put the buttons in some other message

Open to any opinions.

### CODE REVIEW

**Comment by dagar** [comment]
Are you aware of the split between `manual_control_setpoint` and `manual_control_switches`?

https://github.com/PX4/PX4-Autopilot/blob/main/msg/ManualControlSwitches.msg

The idea was to keep the continuous stick changes in the `manual_control_setpoint`, and capture the discrete changes (with additional checks, hysteresis, etc) in `manual_control_switches`.

Then in the case of RC we've got some logic to help map buttons to switches (RC_MAP_FLTM_BTN), but that's bypassed for the Mavlink input case.

tl;dr there's work to do here, but if you need a short term solution maybe simply moving the buttons to `manual_control_switches` would be more tolerable? I would also just do an array of booleans to keep it simple.

**Comment by tstastny** [comment]
Was not aware! But assumed I was not the first to look into this. Thanks for the pointer. No real rush from my side. Just put this up to get the discussion started mainly. 

On RC_MAP_FLTM_BTN, what if the idea is not to change a flight mode but just trigger some aspect of it? Eg like toggle a cruise control setpoint?

@dagar What's your long term proper approach to handling this architecturally? I don't mind the extra leg work.

The hacky way I did this last week was with another param for the specific cruise controller to specify what button mapped to that toggle switch. But of course that does not then block the button from any other use throughout the system.

**Comment by MaEtUgR** [comment]
I'd totally move that to the switches. Sorry my input on a possible implementation was not so detailed from mobile 😇

> long term proper approach to handling this architecturally?

In my eyes, we need a system that allows mapping inputs to trigger actions. E.g. An RC switch position mapped to trigger the camera or switch mode or release a gripper. But the same system can be used to trigger things from MAVLink, Joystick. It doesn't scale to explicitly implement out every single combination of input and action it can trigger.
